src/main/java/net/minecraftforge/client/GuiIngameForge.java

```diff
@@ -253,7 +253,7 @@ private void renderHelmet(ScaledResolution res, float partialTicks)
 
         ItemStack itemstack = this.mc.player.inventory.armorItemInSlot(3);
 
-        if (this.mc.gameSettings.thirdPersonView == 0 && itemstack != null && itemstack.getItem() != null)
+        if (this.mc.gameSettings.thirdPersonView == 0 && itemstack.getItem() != Item.getItemFromBlock(Blocks.AIR))
```

Comment:
Your fix is a literal translation of the old logic, but `!itemstack.isEmpty()`  would be a better check here.